x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/apikey/CrossClusterApiKeyRoleDescriptorBuilder.java

```diff
@@ -18,14 +18,19 @@
 import java.io.IOException;
 import java.util.Arrays;
 import java.util.List;
+import java.util.stream.Stream;
 
 import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
 
 public class CrossClusterApiKeyRoleDescriptorBuilder {
 
-    public static final String[] CCS_CLUSTER_PRIVILEGE_NAMES = { "cross_cluster_search" };
+    // monitor_enrich is needed for ES|QL + ENRICH and https://github.com/elastic/elasticsearch/issues/106926 is related
+    public static final String[] CCS_CLUSTER_PRIVILEGE_NAMES = { "cross_cluster_search", "monitor_enrich" };
```

Comment:
I think i can remove the need to explicitly add this by changing how ES|QL calls has_privileges and will follow up in a seperate PR.